### PR TITLE
[SCIM-20] retry and refresh token in case of unauthorized response

### PR DESCRIPTION
**Pull request by andrea-patricelli**

### CODE REVIEW

File: src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java [hunk 1]
```diff
@@ -98,7 +101,11 @@ protected WebClient getWebclient(final String path, final Map<String, String> pa
     }
 
     protected String getBearerToken() {
-        if (StringUtil.isNotBlank(config.getBearerToken())) {
+        return getBearerToken(false);
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java:104]: Is it really needed to have this `getBearerToken()` with no arguments? Can't we just find callers and replace them with `getBearerToken(false);` ?

File: src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java [hunk 2]
```diff
@@ -136,7 +143,9 @@ protected JsonNode doGet(final WebClient webClient) {
         JsonNode result = null;
 
         try {
-            Response response = webClient.get();
+            Response response = executeAndRetry((token) -> StringUtil.isNotBlank(token)
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java:146]: it seems that all the invocations to `executeAndRetry` replicate the same boilerplate: can't we just embed that in the `executeAndRetry()` method itself?

File: src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java [hunk 3]
```diff
@@ -177,8 +186,10 @@ protected void doCreate(final UT user, final WebClient webClient) {
                 // no custom attributes
                 payload = SCIMUtils.MAPPER.writeValueAsString(user);
             }
-            response = webClient.post(payload);
-
+            LOG.info("CREATE payload is {0}: ", payload);
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java:189]: turn that to `LOG.ok()`

File: src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java [hunk 4]
```diff
@@ -223,17 +234,24 @@ protected JsonNode doUpdate(final UT user, final WebClient webClient) {
                 payload = SCIMUtils.MAPPER.writeValueAsString(user);
             }
 
+            LOG.info("UPDATE payload is {0}: ", payload);
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java:237]: same here and all other places where the whole `payload` is logged

**Review by ilgrosso — CHANGES_REQUESTED** [review]

File: src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java (hunk 1 above)
Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java:104]: Yes, we can. Going to adjust the code accordingly.

File: src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java (hunk 2 above)
Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/scim/common/service/AbstractSCIMService.java:146]: This way how to call different `webClient` http methods without replicating the code?

**Review by ilgrosso — APPROVED** [review]

**Pull request merged**